### PR TITLE
Adding the new 3.0beta2

### PR DESCRIPTION
**Pull request by BrunoVernay**
Hi,
Not sure if a branch is the solution, but I am starting the conversation at least.
- I removed PortAudio and PortMidi (they are more confusing than helpful this days.)
- I enabled Jack (for the Musicians out there)
- I know the appdata.xml  could  be improved
- I added a `branch: beta` in the yaml. I think the default is `master` which may be confusing. It might be good to specify `stable` on the current 2.3.2.

This is my first try at FlatPack ...

### CODE REVIEW

File: org.musescore.MuseScore.appdata.xml [hunk 1]
```diff
@@ -92,7 +92,7 @@
   </screenshots>
   <content_rating type="oars-1.1"/>
   <releases>
-    <release date="2018-07-31" version="2.3.2">
+    <release date="2018-12-07" version="3.0beta2">
```
Comment (jurf) [inline org.musescore.MuseScore.appdata.xml]: Don’t rewrite the old release notes tag with the new one… not only we’ll lose the old release notes, the new ones will be wrong.

File: org.musescore.MuseScore.yml [hunk 2]
```diff
@@ -44,49 +45,33 @@ cleanup-commands:
...
 
-  - name: portaudio
-    config-opts:
-      - --disable-static
-      - --without-oss
-    sources:
-      - type: archive
-        url: http://www.portaudio.com/archives/pa_stable_v190600_20161030.tgz
-        sha256: f5a21d7dcd6ee84397446fa1fa1a0675bb2e8a4a6dceb4305a8404698d8d1513
+  - name: jack2
```
Comment (jurf) [inline org.musescore.MuseScore.yml]: 1. PortMIDI isn’t irrelevant, it enables real-time MIDI input _without_ JACK.

2. I don’t think JACK actually works in Flatpak, it certainly didn’t last time I checked.

File: org.musescore.MuseScore.yml [hunk 3]
```diff
@@ -44,49 +45,33 @@ cleanup-commands:
...
   - name: musescore
     buildsystem: cmake
     config-opts:
       - -DCMAKE_BUILD_TYPE=RELEASE
+      - -DBUILD_PORTAUDIO=OFF
+      - -DBUILD_PORTMIDI=OFF
       - -DBUILD_ALSA=OFF
-      - -DBUILD_JACK=OFF
+      - -DBUILD_JACK=ON
+      - -DUSE_SYSTEM_FREETYPE=ON
```
Comment (jurf) [inline org.musescore.MuseScore.yml]: Don’t combine random stuff into one commit/PR. There was a time when I used this flag, but really the font rendering is much worse without it, so I ultimately decided against it.

File: .gitignore [hunk 4]
```diff
@@ -2,3 +2,4 @@
 appdir/
 repo/
 *.flatpak
+build-dir
```
Comment (jurf) [inline .gitignore]: Why? `appdir` is already there for that purpose.

File: org.musescore.MuseScore.yml [hunk 5]
```diff
@@ -1,4 +1,5 @@
 app-id:              org.musescore.MuseScore
+branch:              beta
```
Comment (jurf) [inline org.musescore.MuseScore.yml]: Flathub implies `stable`.

If you build it on your machine, it’s going to build as `master`, which is useful too, allowing you to install your local version alongside the Flathub one.

Flathub also requires `stable`, so we can’t do this. We also can’t ship beta-quality software on Flathub.

What you’re doing here is suggesting replacing the current stable MuseScore with the unstable beta one. There are cases where this makes sense, especially in old or poorly-maintained projects, but definitely not here.

The best solution would be to create a git branch and merge it when it’s ready.

**Comment by jurf** [comment]
> * I know the appdata.xml  could  be improved

There’s actually a PR for that upstream (MuseScore/MuseScore#4070), so you probably want to make your comments there.

**Comment by jurf** [comment]
But thanks for your work, I planned to try packaging MuseScore 3 for quite some time now, but couldn’t because of school.

**Comment by BrunoVernay** [comment]
Thanks for all your comments.

- I thought that Jack worked because I saw other projects https://github.com/flatpak/flatpak/issues/1509#issuecomment-375304230  using it.  Obviously I should have tested it ...
- Did not no about the master, stable default and flathub's requirements (my idea was to make it available to more users)
- `build-dir` is the directory they use in the Flatpak tutorial.

- FreeType or not does not make such a difference indeed. I had never compared the 2 before. 

I will setup a Google alert on PipeWire and wait.  It is sad, since Flatpak is way easier to build than RPM!

**Comment by jurf** [comment]
@BrunoVernay no need to close the PR! MuseScore 3 will come out sooner than later and this package will need to be updated anyway.

Regarding Flathub’s policies, you can read up on the [Wiki](https://github.com/flathub/flathub/wiki) if you want to, well worth a read if you plan to stay around.

Quick note to the rest – sure other directories will be used elsewhere, though I think currently the docs just use `<build-directory>`, but it was a completely unnecessary change for this PR. Also, it’s not about using the Freetype or not, it’s about using the one from the runtime or the build-in one. And no need to wait for PipeWire, MuseScore works just fine without JACK.

As you can see above, if you want to test MuseScore 3, you can download the build artifacts and install them. Or you can export a Flatpak bundle and provide that to the devs (actually, if you won’t do that, I probably will).

Oh and sorry if my comments were a bit harsh…

**Pull request closed without merging**